rust/parquet/src/arrow/schema.rs

```diff
@@ -71,6 +125,13 @@ where
         .map(|encoded| get_arrow_schema_from_metadata(&encoded))
         .unwrap_or_default();
 
+    // add the Arrow metadata to the Parquet metadata
+    if let Some(arrow_schema) = &arrow_schema_metadata {
```

Comment:
we weren't preserving metadata. Arrow keys will now overwrite existing Parquet keys in the metadata